runner/build.rs

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use platforms::{platform::Platform, target::OS};
+
+fn main() {
```

Comment:
I am still not a big fan of build.rs files in general, why would the same logic not work as part of main.rs?

Reply:
IIUC your other comment, it is just to have a separation so that adding a new platform only requires modifying build.rs and not main.rs? If that's the only reason, IMO it does not justify the complexity of the build script and passing arguments as strings over env variables. The same logic may (and should) still be encapsulated in main.rs so that someone just wanting to add a new platform would be able to do so without modifying (or understanding) the rest of the file.

Reply:
Its peculiar that we have quite opposing views from the similar reasoning! :
The use of build.rs for me is a reduction in complexity of main.rs, and represents a reduction in cludginess of encoding platform strings in the functional code. It separates the platform adaptions cleanly from the functional code, and by using a separate file it helps de-conflict unrelated merges of code/platform hacking (it would keep me out of main.rs ;-). The code uses the environmental variables to help avoid needing to use once-cell type patterns or globals in a similar way, which to my mind further increases complexity (and decreases readability). Conversely, If the code in was imported to main.rs, the volume of code remains would probably actually increase a little, along with the dependencies lists for that module.
I was trying to save everyone a few headaches in future, deconflict development activities and aid the reabability :-)

All this said:

I'll happily put the code in main.rs, your codebase, your rules :-)

Updated patch incoming, I had to also remove the system NPM implementation as is was conflicting... more changes ahoy....

Reply:
Ah I see, but I don't think the once-cell optimizations are necessary, a method that re-computes the string every time is more than fine, I don't think we should worry about a few extra string allocations in a dev tool. Does that seem a reasonable compromise to you?

Reply:
All works for me: I almost had it completely cleanly solved once, then ran into the 'can't implement a trait from one crate on a struct from another' rule, which scuppered that design too... I'll beat it into submission one day ;-) thanks for the tolerance, Cheers !

Reply:
I think what I'm suggesting boils down to just moving the logic from this file to two methods in main.rs: `get_server_rust_target()` and `get_client_rust_target()`, and just invoke them as many times as necessary throughout the code, no need to worry about trying to do it exactly once. Also no need to make the structopt default line up, I think it's fine to make that an Option, and either use the provided value (if any), or if not, rely on the one comeing from the appropriate aforementioned method.

Reply:
Also if you prefer, feel free to put these in a separate platform.rs file, and include that from main.rs, to keep the logic separate, up to you.